client/src/utils/auth.ts

```diff
@@ -54,50 +59,59 @@ export function logoutUser() {
   clearAuthToken();
 }
 
-export async function storeAuthUser(token): Promise<boolean> {
-  setAuthToken(token);
-  const decodedToken = getUserInfo();
-  if (decodedToken !== null) {
-    await store.dispatch('setUser', {
-      email: decodedToken.email,
-      firstName: decodedToken.firstName,
-      id: decodedToken.id,
-      lastName: decodedToken.lastName,
-      sub: decodedToken.sub,
-    });
-    return true;
-  }
-  return false;
+export async function storeAuthUser(decodedToken) {
+  await store.dispatch('setUser', {
+    email: decodedToken.email,
+    firstName: decodedToken.firstName,
+    id: decodedToken.id,
+    lastName: decodedToken.lastName,
+    sub: decodedToken.sub,
+  });
 }
 
-export async function storeAuthUserProfile(userProfile) {
+async function storeAuthUserProfile(userProfile) {
   await store.dispatch('setUserProfile', {
     ...userProfile,
   });
 }
 
 export async function getAuthUserProfile() {
   try {
-    const res = await http.get('/auth/profile');
-    await storeAuthUserProfile(res.data);
-  } catch (err) {
-    await store.dispatch('addNotification', { text: `Fehler beim Laden des Nutzerprofils [ERROR: ${err.message}]`, type: 'danger' });
+    return await http.get('/auth/profile');
+  } catch (err: any) {
+    throw new Error(`Fehler beim Laden des Nutzer:innenprofils [ERROR: ${err.message}]`);
   }
 }
 
 export async function login(username: string, password: string): Promise<boolean> {
+  let res: AxiosResponse;
+  try {
+    res = await http.post('/auth/login', { username, password });
+  } catch (err: any) {
+    await showErrorNotification(`Falsche Emailadresse oder Passwort: ${err.message}`);
```

Comment:
```suggestion
    // TODO: Differentiate between errors
    await showErrorNotification(`Falsche Emailadresse oder Passwort: ${err.message}`);
```
Hier müssten wir vielleicht später noch zwischen verschiedenen Errors unterscheiden (z.B. LDAP nicht erreichbar, 500, ...), damit die Fehlermeldung ein bisschen aussagekräftiger ist.